spi/src/main/java/org/opensearch/security/spi/resources/ResourceSharingExtension.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.security.spi.resources;
+
+/**
+ * This interface should be implemented by all the plugins that define one or more resources.
+ *
+ * @opensearch.experimental
+ */
+public interface ResourceSharingExtension {
```

Comment:
nit: Similar to previous comment, wdyt about naming this `SharableResourceExtension`?

Reply:
IMO ResourceSharingExtension is more clear.